pkg/model/api/v1/dashboard.go

```diff
@@ -27,7 +27,7 @@ import (
 type PanelSpec struct {
 	Display common.Display `json:"display" yaml:"display"`
 	Plugin  common.Plugin  `json:"plugin" yaml:"plugin"`
-	Queries []Query  `json:"queries" yaml:"queries"`
+	Queries []Query        `json:"queries,omitempty" yaml:"queries, omitempty"`
```

Comment:
Is there a semantic difference between `queries: []` vs no queries? Double checking to make sure `omitempty` is what we expect

Reply:
Gotcha! That seems reasonable to me, but I'll defer to @Nexucis to 👍  the backend-y bits, since it's less my area of expertise.

Reply:
There is different thing behind the tag `omitempty` in Go.

Technically speaking about what does it do:
- when the struct will be marshalled in JSON, if the array is empty then it won't appear in the final JSON.
- when the JSON is unmarshalled in the struct, if the key doesn't appear then the array is not initialised in the struct.

Sementically speaking, and because of the behaviour described above, I think that's a nice way to indicate this field is optional.